### PR TITLE
Fix #9900 - Conditions doesn't recognize some of the characters set

### LINKED ISSUE #9900: Workflow - Conditions doesn't recognize some of the characters sets
When having a condition of a text field that contains any special character, such as accents, "ñ", or "c" characters, the condition doesn't fulfill and the workflow isn't processed.
HtmlCondition

This is because the workflow doesn't decode profile the HTML entities

#### Issue
Workflow conditions of text fields that compare special characters doesn't work properly

#### Expected Behavior
It should work with any character.

#### Actual Behavior
The workflows isn't executed cause the condition isn't met.

#### Possible Fix
Add the code for decode HTML entities.

#### Steps to Reproduce

1. Create a workflow that compares "aña" with last_name of a Contact
2. Add any action
3. Create a Contact with last_name "aña" and check the the Workflow didn't execute

#### Context
It should work with any character.

#### Your Environment
* SuiteCRM Version used: 7.12.8
* Browser name and version: Versión 107.0.5304.121 (Build oficial) (64 bits)
* Environment name and version: MariaDB 10.6.5, PHP 7.4
* Operating System and version : KDE Neon 5.26

### PR DESCRIPTION
**Pull request by SinergiaCRM**
- Closes #9900 

## Description
This PR adds the code that decode the HTML value of a text field in a Workflow condition. Therefore any character set can be used in condition

## Motivation and Context
Condition should work with any special character.

## How To Test This
1. Create a workflow that compares "aña" with last_name of a Contact
2. Add any action
3. Create a Contact with last_name "aña" and check the the Workflow run successfully.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [x] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ ] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

### CODE REVIEW

File: modules/AOW_WorkFlow/AOW_WorkFlow.php [hunk 1]
```diff
@@ -823,6 +823,10 @@ public function check_valid_bean(SugarBean $bean)
                         } elseif ($data['type'] == 'bool' && (!(bool)$value || strtolower($value) == 'false')) {
                             $value = 0;
                         }
+                        $type = $data['dbType'] ?? $data['type'];
+                        if ((strpos($type, 'char') !== false || strpos($type, 'text') !== false) && !empty($field)) {
+                            $field = from_html($field);
```
[the remark this answers is not in the document]
Comment (AlbertoSTIC) [reply modules/AOW_WorkFlow/AOW_WorkFlow.php:828]: Hi @clemente-raposo ,

We use `from_html` cause this is used also in the same file:
https://github.com/salesagility/SuiteCRM/blob/f4d77b5bd5eaa43e854aeb2ea7c0ae6f9d1c3e30/modules/AOW_WorkFlow/AOW_WorkFlow.php#L722-L724

But we can change it if you think it's more convenient.

[the remark this answers is not in the document]
Comment (clemente-raposo) [reply modules/AOW_WorkFlow/AOW_WorkFlow.php:828]: Hi @AlbertoSTIC,

Thank you. Make sense, lets keep the `from_html`

**Review by clemente-raposo — APPROVED** [review]
LGTM

**Review by johnM2401 — APPROVED** [review]
LGTM, Checked each of the Workflow Condition operators, all work as expected.

**Pull request merged**